repos/system_upgrade/common/libraries/rpms.py

```diff
@@ -65,6 +65,21 @@ def get_installed_rpms():
         return []
 
 
+def parse_rpm_details(rpm):
+    """
+    Parse rpm details. In case of failure, log a warning and return None.
+
+    :param rpm: string with rpm details separated by '|'
+    """
+
+    try:
+        name, version, release, epoch, packager, arch, pgpsig = rpm.split('|')
+        return name, version, release, epoch, packager, arch, pgpsig
+    except ValueError:
+        stdlib.api.current_logger().warning('Could not parse rpm: {}, skipping.'.format(rpm))
+        return None
+
+
```

Comment:
@tomasfratrik I do not like this function in the shared library. first, we know already we will have to change the separator / format, which will make this function problematic then. second, you have to input the string in very exact format! not just separated by '|', but the number of items/chunks and order must be very specific as well. the format is not described here, there is not function to obtain such a string, so I do not consider this good enough to publish anything like that. this is not good design basically.

Reply:
That is a great point that no function has as an output what this function has as parameter. And since separator will be changed, there is not point of having that just for now. So I'm removing it.